### ISSUE TITLE
Suppress scan of classes annotated with @Generated annotation

### ISSUE THREAD

**Issue by davido** [google/error-prone#505]
I'm using Error Prone 2.0.15, im context of this PR: facebook/buck#1047.

Shouldn't the classes annotated with @Generated annotation be ignored per default? This doesn't seem to work for Buck code base, for example for this generated class:

```
@SuppressWarnings({"all"})
@ParametersAreNonnullByDefault
@Generated({"Immutables.generator", "AbstractApplePlatform"})
@Immutable
@CheckReturnValue
public final class ApplePlatform extends AbstractApplePlatform {
[...]
```

this error and other warnings are reported:

```
/home/travis/build/facebook/buck/buck-out/annotation/src/com/facebook/buck/apple/__platform_gen__/com/facebook/buck/apple/ApplePlatform.java:333: error: [OptionalEquality] Comparison using reference equality instead of value equality
    if (this.stubBinaryPath == value) return this;
                            ^
    (see http://errorprone.info/bugpattern/OptionalEquality)
  Did you mean 'if (Objects.equals(this.stubBinaryPath, value)) return this;' or 'if (this.stubBinaryPath.equals(value)) return this;'?
```

The full Travis log is here: https://travis-ci.org/facebook/buck/builds/189866663 .

**Comment by eaftan** [comment]
We distinguish between errors and warnings, and only disable warnings in generated code.  The idea was that anything important enough to be an error should not appear in generated code.

See the discussion in #329.

**Comment by davido** [comment]
As I mentioned in the description above, the warnings weren't suppressed either:

```
/home/travis/build/facebook/buck/buck-out/annotation/src/com/facebook/buck/apple/__platform_gen__/com/facebook/buck/apple/ApplePlatform.java:333: warning: [ReferenceEquality] Comparison using reference equality instead of value equality
    if (this.stubBinaryPath == value) return this;
                            ^
    (see http://errorprone.info/bugpattern/ReferenceEquality)
  Did you mean 'if (Objects.equals(this.stubBinaryPath, value)) return this;' or 'if (this.stubBinaryPath.equals(value)) return this;'?
```

Or is this because existence of errors?

**Comment by davido** [comment]
Yes, this is the case.
